src/server/node_manager.c

```diff
@@ -6850,124 +6850,100 @@ set_nodes(void *pobj, int objtype, char *execvnod_in, char **execvnod_out, char
 void
 free_nodes(job *pjob)
 {
-	struct	pbssubn	*np;
-	mom_svrinfo_t	*psvrmom;
-	struct  pbsnode *pnode;
-	struct	jobinfo	*jp, *prev, *next;
-	int     i;
-	int     j;
-	int     ivnd;
-	int     still_has_jobs;	/* still jobs on this vnode */
-	int     special_case = 0;
 
-	DBPRT(("%s: entered\n", __func__))
-
-	/* decrement number of jobs on the Mom who is the first Mom */
-	/* for the job, Mother Superior; incremented in set_nodes() */
-	/* and saved in ji_destin in assign_hosts()		    */
-	if (((pjob->ji_qs.ji_svrflags & JOB_SVFLG_HasNodes) != 0) &&
-		(pjob->ji_qs.ji_destin[0] != '\0')) {
-		pnode = find_nodebyname(pjob->ji_qs.ji_destin);
-		if (pnode) {
-			psvrmom = pnode->nd_moms[0]->mi_data;
-			if (--psvrmom->msr_numjobs < 0)
-				psvrmom->msr_numjobs = 0;
-		}
+	struct pbsnode *pnode;
+	mom_svrinfo_t *psvrmom;
+	char *execvnod_in = NULL;
+	int i;
+	int j;
+	char *execvncopy;
+	char *chunk;
+	char *last;
+	int hasprn;
+	char *vname;
+	int nelem;
+	struct key_value_pair *pkvp;
+	char *execvnod = NULL;
+	int special_case = 0;
+	
+	/* decrement number of jobs on the Mom who is the first Mom */	
+	/* for the job, Mother Superior; incremented in set_nodes() */	
+	/* and saved in ji_destin in assign_hosts()		    */	
+	if (((pjob->ji_qs.ji_svrflags & JOB_SVFLG_HasNodes) != 0) &&	
```

Comment:
this can be part of deallocate_job_from_node instead of handling separately.

Reply:
Yes, that is what I am trying now.

Reply:
@bremanandjk Looks like you missed push this change?

Reply:
On debugging further, I have realized we should not add this code in deallocate_job_from_node(). So left this code as it is.